src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -287,22 +289,51 @@ private void failUnpublished(final List<BatchItem> batch, final String reason) {
     @Override
     public Optional<PartitionStatistics> loadPartitionStatistics(final Timeline timeline, final String partition)
             throws ServiceUnavailableException {
+        return loadPartitionStatistics(Collections.singletonList(new TimelineAndPartition(timeline, partition))).get(0);
+    }
+
+    @Override
+    public List<Optional<PartitionStatistics>> loadPartitionStatistics(
```

Comment:
I do not see an easy way of doing this without loosing readability or speed of methods (I mean `consumer.partitionsFor` call.